packages/framework-types/src/envelope.ts

```diff
@@ -137,3 +138,8 @@ export interface UserEnvelope {
   role: string
   claims: Record<string, unknown>
 }
+
+export interface RequestEnvelope {
+  headers: unknown
+  body: unknown
```

Comment:
Let's not restrict this only to the header and the body of the request, but include all the context we have, including data from the cloud provider. If we want to highlight some important fields with types it's ok, but I'd add a placeholder to access other fields if necessary. I'd rather have the entire context than only the header and the body:

```suggestion
  request: {
    headers: unknown
    body: unknown
    [key: string] : unknown
  }
  [key: string] : unknown
```

Reply:
@javiertoledo The reason behind the constraint of only adding the header and the body is because the context might be quite long and I don't know if that could cause any problem. Makes sense?

Reply:
Fixed in d43ea9cb9848e447a39abaa483e34388613d0524